tileview/src/main/java/com/qozix/tileview/tiles/TileRenderPoolExecutor.java

```diff
@@ -0,0 +1,113 @@
+package com.qozix.tileview.tiles;
+
+import java.lang.ref.WeakReference;
+import java.util.LinkedList;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.Future;
+import java.util.concurrent.LinkedBlockingDeque;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import android.os.Process;
+
+public class TileRenderPoolExecutor {
+  // Sets the amount of time an idle thread will wait for a task before terminating
+  private static final int KEEP_ALIVE_TIME = 1;
+  private static final TimeUnit KEEP_ALIVE_TIME_UNIT = TimeUnit.SECONDS;
+
+  private static final int CORE_POOL_SIZE = Runtime.getRuntime().availableProcessors();
+  private static final int MAXIMUM_POOL_SIZE = Runtime.getRuntime().availableProcessors();
+
+  private final CustomThreadPoolExecutor mExecutor;
+  private final BlockingQueue<Runnable> mQueue;
+  private final BlockingQueue<Future> mFutureList;
+  private WeakReference<TileCanvasViewGroup> mViewGroup;
+
+  private static TileRenderPoolExecutor sInstance = null;
+
+  private TileRenderPoolExecutor() {
+    mQueue = new LinkedBlockingDeque<>();
+    mExecutor = new CustomThreadPoolExecutor( CORE_POOL_SIZE, MAXIMUM_POOL_SIZE, KEEP_ALIVE_TIME, KEEP_ALIVE_TIME_UNIT, mQueue );
+    mFutureList = new LinkedBlockingDeque<>();
+  }
+
+  public static TileRenderPoolExecutor getsInstance() {
```

Comment:
So can we use the on-demand idiom i linked to previous: https://en.wikipedia.org/wiki/Singleton_pattern#Initialization-on-demand_holder_idiom - something like:

```
private static class SingletonHolder {
  private static final TileRenderPoolExecutor INSTANCE = new TileRenderPoolExecutor();
}
public static TileRenderPoolExecutor getInstance() {
  return SingletonHolder.INSTANCE;
}
```

That said, I don't see any reason why this needs to be a singleton at all - if you want to just instantiate an instance in `TileCanvasViewGroup` constructor, that'd be fine as well.